src/socket/udp.rs

```diff
@@ -156,18 +148,44 @@ impl<'a, 'b> UdpSocket<'a, 'b> {
     /// to its payload.
     ///
     /// This function returns `Err(Error::Exhausted)` if the transmit buffer is full,
-    /// `Err(Error::Truncated)` if the requested size is larger than the packet buffer
+    /// `Err(Error::Truncated)` if the requested size is larger than the payload buffer
```

Comment:
It is not necessary to return `Error::Truncated` from this function anymore since there are no separate packet buffers.